### PR TITLE
[NSLOOKUP] Fix crash in case of no network connectivity

### PR DESCRIPTION
**Pull request by kirschju**
Prevent nslookup.exe from crashing when executed in a ROS VM with no network
interfaces. This is due to a NULL pointer dereference occurring if
`GetNetworkParams` in `main` fails with an error other than
`ERROR_BUFFER_OVERFLOW`. In this case, `pNetInfo` remains initialized to
NULL, causing `strncpy` to crash.

## Purpose

Fixes a bug in nslookup.exe causing it to crash with a NULL pointer dereference in situations with no network connectivity.

JIRA issue: *none*

## Proposed changes

Move the entire logic that copies the results obtained from `GetNetworkParams` to the global `State` struct into the branch where the first `GetNetworkParams` call was successful but reported the desired size for `pNetInfo`. This additionally is a bit more consistent with regard to `pNetInfo` being allocated and freed within the same code path.

### CODE REVIEW

File: base/applications/network/nslookup/nslookup.c [hunk 1]
```diff
@@ -812,15 +812,16 @@ int main( int argc, char* argv[] )
...
-             pNetInfo->DnsServerList.IpAddress.String,
-             15 );
+        strncpy( State.domain, pNetInfo->DomainName, 255 );
+        strncpy( State.srchlist[0], pNetInfo->DomainName, 255 );
+        strncpy( State.DefaultServerAddress,
+                 pNetInfo->DnsServerList.IpAddress.String,
+                 15 );
+
+        HeapFree( ProcessHeap, 0, pNetInfo );
+    }
```
Comment (Extravert-ir) [inline base/applications/network/nslookup/nslookup.c]: Shouldn't we just exit the program if GetNetworkParams returned something else than ERROR_BUFFER_OVERFLOW?

Comment (kirschju) [reply base/applications/network/nslookup/nslookup.c]: Probably a sane decision. I have a hard time imagining a scenario in which it is possible to send DNS requests, but retrieving the network parameters of the *local* machine fails. Please see the new version of the fix.

File: base/applications/network/nslookup/nslookup.c [hunk 2]
```diff
@@ -792,26 +792,30 @@ int main( int argc, char* argv[] )
     /* We don't know how long of a buffer it will want to return. So we'll
        pass an empty one now and let it fail only once, instead of guessing. */
     Status = GetNetworkParams( pNetInfo, &NetBufLen );
-    if( Status == ERROR_BUFFER_OVERFLOW )
+
+    if( Status != ERROR_BUFFER_OVERFLOW ) {
```
Comment (ThFabba) [inline base/applications/network/nslookup/nslookup.c]: The rest of this file appears to put braces on their own lines. Would be good to stay consistent.

File: base/applications/network/nslookup/nslookup.c [hunk 3]
```diff
@@ -822,6 +826,7 @@ int main( int argc, char* argv[] )
 
     HeapFree( ProcessHeap, 0, pNetInfo );
 
+
```
Comment (ThFabba) [inline base/applications/network/nslookup/nslookup.c]: This seems unnecessary.

File: base/applications/network/nslookup/nslookup.c (hunk 2 above)
Comment (kirschju) [reply base/applications/network/nslookup/nslookup.c]: True. Fixed in the updated version.

File: base/applications/network/nslookup/nslookup.c (hunk 3 above)
Comment (kirschju) [reply base/applications/network/nslookup/nslookup.c]: Yes. Sorry for being inconsistent. Also updated in the latest version.

**Pull request merged**